lib/clients/gitlab.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'rest-client'
+require 'json'
+
+module Octopus
+  module Clients
+    class Gitlab
+      API_PATH = 'api/v4'
+      attr_reader :base_url, :username, :password
+
+      def initialize(base_url, username, password)
+        @base_url = base_url
+        @username = username
+        @password = password
+      end
+
+      # What Bitbucket calls Projects/Repos, Gitlab and Github call Groups/Projects.
+      # For now, maintain compatibility with Bitbucket.
+      def projects(limit = 1000)
+        url = "#{base_url}/#{API_PATH}/groups"
+        response = RestClient::Request.execute(method: :get, url: url, headers: { private_token: password })
+        JSON.parse(response.body).first(limit).each do |group|
+          # For now, maintain compatibility with Bitbucket.
+          group['key'] = group.delete('path')
+        end
+      end
+
+      def repos(project_key, limit = 1000)
```

Comment:
I think that since there is no implementation for these methods, they should at least temporarily `raise "Not implemented yet."` or something like that.

Also, we might be better of by extracting these interface-like methods into a separate base class and have them approximately like so:
```ruby
class MyBaseClient
  def repos(project_key, limit = 1000)
    raise 'Must be implemented in a child class.'
  end
end

class MyScmClient < MyBaseClient
  def repos(project_key, limit = 1000)
    RestClient.get(url, ... )
  end
end
```

This will allow us to clearly define interface methods that we expect each client to implement, and make the client implementation more succinct in its early stages.